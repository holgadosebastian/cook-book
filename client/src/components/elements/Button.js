import React from 'react';
import PropTypes from 'prop-types';

const Button = ({
  color = 'primary',
  outlined,
  rounded = true,
  loading,
  cssClasses,
  onClick,
  children,
  ...rest
}) => {
  return (
    <button
      {...rest}
      className={`button is-${color} ${
        rounded && 'is-rounded'
      } is-uppercase ${cssClasses} ${loading && 'is-loading'} ${
        outlined && 'is-outlined'
      }`}
      onClick={onClick}
    >
      {children}
    </button>
  );
};

Button.propTypes = {
  color: PropTypes.string,
  outlined: PropTypes.bool,
  rounded: PropTypes.bool,
  loading: PropTypes.bool,
  cssClases: PropTypes.string,
  onClick: PropTypes.func
};

export default Button;
