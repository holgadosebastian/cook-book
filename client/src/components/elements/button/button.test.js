import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Button from './button';

describe('Button Component', () => {
  it('Should exist', () => {
    expect(Button).toMatchSnapshot();
  });

  it('Should render a default button', () => {
    const button = renderer.create(<Button>Click</Button>);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('Should render a secondary button', () => {
    const button = renderer.create(<Button color='secondary'>Click</Button>);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('Should render a full width button', () => {
    const button = renderer.create(<Button fullwidth>Click</Button>);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('Should render an outlined button', () => {
    const button = renderer.create(<Button outlined>Click</Button>);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('Should render a loading button', () => {
    const button = renderer.create(<Button loading>Click</Button>);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('Should render a disabled button', () => {
    const button = renderer.create(<Button disabled>Click</Button>);
    expect(button.toJSON()).toMatchSnapshot();
  });

  it('Should have a click listener', () => {
    const onClick = jest.fn();
    const button = shallow(<Button onClick={onClick} />);
    button.simulate('click');
    button.simulate('click');
    expect(onClick).toHaveBeenCalledTimes(2);
  });

  it('Should not dispatch click event if disabled', () => {
    const onClick = jest.fn();
    const button = shallow(<Button disabled onClick={onClick} />);
    button.simulate('click');
    expect(onClick).toHaveBeenCalledTimes(0);
  });

  it('Should have a default click event', () => {
    const spy = jest.spyOn(Button.defaultProps, 'onClick');
    const button = shallow(<Button />);
    button.simulate('click');
    button.simulate('click');
    expect(spy).toHaveBeenCalledTimes(2);
    Button.defaultProps.onClick.mockRestore();
  });
});
