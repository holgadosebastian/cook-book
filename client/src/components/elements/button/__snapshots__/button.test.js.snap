// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should exist 1`] = `[Function]`;

exports[`Button Component Should render a default button 1`] = `
<button
  className="button is-uppercase is-primary is-rounded"
  disabled={false}
  onClick={[Function]}
>
  Click
</button>
`;

exports[`Button Component Should render a disabled button 1`] = `
<button
  className="button is-uppercase is-primary is-rounded"
  disabled={true}
>
  Click
</button>
`;

exports[`Button Component Should render a full width button 1`] = `
<button
  className="button is-uppercase is-primary is-rounded is-fullwidth"
  disabled={false}
  onClick={[Function]}
>
  Click
</button>
`;

exports[`Button Component Should render a loading button 1`] = `
<button
  className="button is-uppercase is-primary is-rounded is-loading"
  disabled={false}
  onClick={[Function]}
>
  Click
</button>
`;

exports[`Button Component Should render a secondary button 1`] = `
<button
  className="button is-uppercase is-secondary is-rounded"
  disabled={false}
  onClick={[Function]}
>
  Click
</button>
`;

exports[`Button Component Should render an outlined button 1`] = `
<button
  className="button is-uppercase is-primary is-rounded is-outlined"
  disabled={false}
  onClick={[Function]}
>
  Click
</button>
`;
